### ISSUE TITLE
Investigate need for a `.p-strip--hero` pattern

### ISSUE THREAD

**Issue by barrymcgee** [vanilla-framework/vanilla-brochure-theme#13]

**Comment by yaili** [comment]
@barrymcgee @anthonydillon I don't remember this pattern and don't see it in meeting notes. Can you add some detail?

**Comment by anthonydillon** [comment]
I believe we thought the heros may have different padding from the standard
strip.

**Comment by barrymcgee** [comment]
@yaili Yes - the hero in the `vanillaframework.io` design has a larger padding than any other strip.

**Comment by yaili** [comment]
I think that should be an option for any strip, not just be constrained to heros.

**Comment by anthonydillon** [comment]
Would you mind specing out the strip modifiers?

**Comment by yaili** [comment]
@anthonydillon what was going to be the spec for strip hero?

**Comment by anthonydillon** [comment]
Nothing if you don't think there is any differences.

**Comment by barrymcgee** [comment]
So.. we've decided we don't need this? Will I close?

**Comment by yaili** [comment]
I couldn't find in the notes the spec for this or the decision that we were going to do it. Do you have that info somewhere?

**Comment by barrymcgee** [comment]
@yaili All discussion on it has been in this thread above...

**Comment by yaili** [comment]
OK. In that case we need to compare the different sites that will use the theme and see what should be padding for a hero strip based on existing values, not just ubuntu.com. I won't be able to do this before the 27th though.

**Comment by yaili** [comment]
Did some quick investigation just to get some numbers. So here's existing vertical padding values on sites who'll use the brochure theme:

ubuntu.com:
- Row: top 100px, bottom 100px
- Hero row h1s have an extra 20px top padding

canonical.com:
- On homepage, rows have combined padding 100px, some 120px
- Hero rows: top 40px, bottom 60px
- Most other rows: top 60px, bottom 40px (extra 20px added by columns within rows)

maas.io:
- Row: top 60px, bottom 40px
- Homepage hero: top combined 60px, bottom combined 80px
- Various elements inside rows have applied class to add extra top/bottom 30px padding
- Cols inside rows have margin bottom 20px
- Effective vertical row padding is many rows is closer to 90-100px, but 60px/40px is also used

landscape.canonical.com:
- Row: top 60px, bottom 40px
- Homepage hero: top 80px

In summary, it seems there are two main sizes of vertical padding: 60px, or 100px, but neither of these are exclusive to hero rows.

Should we consider having 2 types of row padding:
- narrow/compact/small
- loose/wide/large
(maybe we can think of better words for this)

Thoughts? @spencerbygraves @ kwm14 @joanasa89 @barrymcgee @anthonydillon

**Comment by anthonydillon** [comment]
I prefer the words row-wide/row/row-compact.

**Comment by spencerbygraves** [comment]
Sounds like a good idea to have 2 types, one that is slightly deeper for hero rows (I assume this means a row that only appears at the top of the page).

**Comment by yaili** [comment]
@spencerbygraves from what've found out looking at the padding of existing brochure theme sites, the hero rows aren't necessarily taller than the other rows, so the proposal would be that any row can be either wide or compact. What do you think?

**Comment by spencerbygraves** [comment]
Ok, makes sense. Perhaps row-shallow and row-deep (or similar) would be better, seeing as we're talking about depth and not width?

**Comment by anthonydillon** [comment]
Just wondering, and tell me to shut up if I'm crazy. But, I thought as we
are talking about labelling strips with depth terms it might be nice to
start thinking of them as z-index depths. For example, deep would jump out
of the page and have move room. Whereas, shallow could be a smaller strip
which inset the page?

**Comment by yaili** [comment]
@anthonydillon do you mean also having inset shadows?

**Comment by anthonydillon** [comment]
Yeah, some layer management. Just a thought.

**Comment by barrymcgee** [comment]
I see this working as a setting - much like animations..

We could provide a SASS map with arbitary scales - https://www.sitepoint.com/better-solution-managing-z-index-sass/

**Comment by anthonydillon** [comment]
I meant more visual z-index so applying shadows to make strips look raised
or inset.

**Comment by yaili** [comment]
Perhaps this is something for the future. I would say that for now we can start with the two types of vertical strip padding that we use more often and propose something more complex once we have this in place. Does that sound OK?

**Comment by yaili** [comment]
I've submitted a proposal, will close this issue as it was to investigate the need for strip hero.
